internal/cmd/detach.go

```diff
@@ -395,15 +407,52 @@ func detachProgram(repo execute.OpenRepoResult, data detachData, finalMessages s
 				Parent: data.config.ValidatedConfigData.MainBranch,
 			},
 		)
-		for _, child := range data.config.NormalConfig.Lineage.Children(data.branchToDetachName, data.config.NormalConfig.Order) {
+	}
+	if proposal, hasProposal := data.branchToDetachProposal.Get(); hasProposal {
+		prog.Value.Add(
+			&opcodes.ProposalUpdateTarget{
+				NewBranch: data.config.ValidatedConfigData.MainBranch,
+				OldBranch: data.parentBranch,
+				Proposal:  proposal,
+			},
+		)
+	}
+	for _, child := range data.children {
+		prog.Value.Add(
+			&opcodes.LineageParentSet{
+				Branch: child.name,
+				Parent: data.parentBranch,
+			},
+		)
+
+		if proposal, hasProposal := child.proposal.Get(); hasProposal {
 			prog.Value.Add(
-				&opcodes.LineageParentSet{
-					Branch: child,
-					Parent: data.parentBranch,
+				&opcodes.ProposalUpdateTarget{
+					NewBranch: data.parentBranch,
+					OldBranch: data.branchToDetachName,
+					Proposal:  proposal,
 				},
 			)
 		}
 	}
+	if data.config.NormalConfig.ProposalsShowLineage == forgedomain.ProposalsShowLineageCLI {
+		_ = sync.AddStackLineageUpdateOpcodes(
+			sync.AddStackLineageUpdateOpcodesArgs{
```

Comment:
Nitpick: The rest of the codebase has the data type for the named arguments on the previous line. This makes it look like proper named arguments. Would that work for you here?

```suggestion
		_ = sync.AddStackLineageUpdateOpcodes(sync.AddStackLineageUpdateOpcodesArgs{
```

Reply:
https://github.com/git-town/git-town/pull/5606/commits/f7dbc98f64d2c2bd2c1c9495337bbe7c01a2c6a4